f/medical/fn_famRagDoll.sqf

```diff
@@ -0,0 +1,67 @@
+// F3 FA Medical
+// Credits and documentation: https://github.com/folkarps/F3/wiki
+// ====================================================================================
+
+// INITIALIZE
+params ["_unit"];
+
+// ====================================================================================
+
+// EVENT HANDLER 
+_unit addEventHandler ["AnimStateChanged", {
+    
+    // EH PARAMS
+	params ["_unit", "_anim"];
+
+    // Detect animations which should put us on our belly.
+    if (_anim == "unconsciousfacedown" || _anim == "unconsciousfaceright" || _anim == "amovppnemstpsnonwnondnon_turnl" || _anim == "amovppnemstpsnonwnondnon") then {
+
+        if (f_param_debugMode == 1) then
+        {
+            systemChat "belly";
+        };
+
+        [_unit, _thisEvent, _thisEventHandler] spawn {
+
+            _unit = _this select 0;
+            _tE = _this select 1;
+            _tEH = _this select 2;
+    
+            _unit setPosWorld getPosWorld player;
+
+            if !(_unit getVariable ["FAM_CONSCIOUS",false]) then {
+
+                // Set preferred animation.
+                _unit playMove "Unconscious"; 
+                _unit switchmove "Unconscious"; 
+
+                // Wait a little for animation to set, then disable simulation to lock player.
+                sleep 2;
+                _unit enableSimulation false;
+
+            };
+            // Remove this eventHandler once it happens.
+            _unit removeEventHandler [_tE, _tEH];
+        };
+    };
+}]; 
+
+// ====================================================================================
+
+
+// this bit avoid a BI bug that removes all actions if you ragdoll while healing.
+if ((animationState _unit == "ainvpknlmstpslaywrfldnon_medic" || animationState _unit == "ainvppnemstpslaywrfldnon_medic") && !(_unit getVariable ["FAM_FLAG",false])) then {
+
+    _unit setCaptive true;
+    private _dmg = damage _unit;
+
+    waitUntil {sleep 0.1; (animationState _unit != "ainvpknlmstpslaywrfldnon_medic" && animationState _unit != "ainvppnemstpslaywrfldnon_medic")};
+
+    player setDamage _dmg;
+    player addItem "FirstAidKit"; //replace the one they won't know they lost.
+    
```

Comment:
A direct unit reference should be used rather than `player` (should it be `_unit` or some other unit?)
If this runs on an AI in this state, it will cause the effects to apply to whichever player the AI is local to, which would be bad.